script.go

```diff
@@ -0,0 +1,581 @@
+			return insertOrDelete(false, db, s, args...)
+		},
+	)
+}
+
+func getTable(db *DB, tableName string) (*AnyTable, ReadTxn, error) {
+	txn := db.ReadTxn()
+	meta := db.GetTable(txn, tableName)
+	if meta == nil {
+		return nil, nil, fmt.Errorf("table %q not found", tableName)
+	}
+	return &AnyTable{Meta: meta}, txn, nil
+}
+
+func insertOrDelete(insert bool, db *DB, s *script.State, args ...string) (script.WaitFunc, error) {
+	if len(args) < 2 {
+		return nil, fmt.Errorf("bad args, expected table and path(s)")
+	}
+
+	tbl, _, err := getTable(db, args[0])
+	if err != nil {
+		return nil, err
+	}
+
+	wtxn := db.WriteTxn(tbl.Meta)
+	defer wtxn.Commit()
+
+	for _, arg := range args[1:] {
+		data, err := os.ReadFile(s.Path(arg))
+		if err != nil {
+			return nil, fmt.Errorf("ReadFile(%s): %w", arg, err)
+		}
+		parts := strings.Split(string(data), "---")
+		for _, part := range parts {
+			obj, err := tbl.UnmarshalYAML([]byte(part))
+			if err != nil {
+				return nil, fmt.Errorf("Unmarshal(%s): %w", arg, err)
+			}
+			if insert {
+				_, _, err = tbl.Insert(wtxn, obj)
+				if err != nil {
+					return nil, fmt.Errorf("Insert(%s): %w", arg, err)
+				}
+			} else {
+				_, _, err = tbl.Delete(wtxn, obj)
+				if err != nil {
+					return nil, fmt.Errorf("Delete(%s): %w", arg, err)
+				}
+
+			}
+		}
+	}
+	return nil, nil
+}
+
+func PrefixCmd(db *DB) script.Cmd {
+	return script.Command(
+		script.CmdUsage{
+			Summary: "Query table by prefix",
+			Args:    "table key (-o=<file>)",
+		},
+		func(s *script.State, args ...string) (script.WaitFunc, error) {
+			return prefixOrLowerbound(false, db, s, args)
+		},
+	)
+}
+
+func LowerBoundCmd(db *DB) script.Cmd {
+	return script.Command(
+		script.CmdUsage{
+			Summary: "Query table by lower bound search",
+			Args:    "table key (-o=<file>)",
+		},
+		func(s *script.State, args ...string) (script.WaitFunc, error) {
+			return prefixOrLowerbound(true, db, s, args)
+		},
+	)
+}
+
+func prefixOrLowerbound(lowerbound bool, db *DB, s *script.State, args []string) (script.WaitFunc, error) {
+	var flags flag.FlagSet
+	file := flags.String("o", "", "File to write results to instead of log")
+	if err := flags.Parse(args); err != nil {
+		return nil, fmt.Errorf("bad args: %s", err)
+	}
+	args = flags.Args()
+	if len(args) < 2 {
+		return nil, fmt.Errorf("bad args, expected table and key")
+	}
+
+	tbl, txn, err := getTable(db, args[0])
+	if err != nil {
+		return nil, err
+	}
+
+	var w io.Writer
+	if *file == "" {
+		w = s.LogWriter()
+	} else {
+		f, err := os.OpenFile(s.Path(*file), os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0644)
+		if err != nil {
+			return nil, fmt.Errorf("WriteFile(%s): %s", *file, err)
+		}
+		defer f.Close()
+		w = f
+	}
+
+	tw := tabwriter.NewWriter(w, 5, 4, 3, ' ', 0)
+	header := tbl.TableHeader()
+	fmt.Fprintf(tw, "%s\n", strings.Join(header, "\t"))
+
+	var it iter.Seq2[any, uint64]
+	if lowerbound {
+		it = tbl.LowerBound(txn, args[1])
```

Comment:
The querying needs a bit more work. Should add support for specifying which index to query and perhaps extend the `Index` type to have a `FromString` method (`func(string) index.Key`) so this works for any type of key and index that implements it.